src/components/calcite-picker-row/calcite-picker-row.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, Element, Event, EventEmitter, Host, Method, Prop, h } from "@stencil/core";
+import {
+  checkSquare16,
+  circle16,
+  circleFilled16,
+  handleVertical24,
+  square16,
+  trash16
+} from "@esri/calcite-ui-icons";
+import { CSS } from "./resources";
+import CalciteIcon from "../utils/CalciteIcon";
+
+@Component({
+  tag: "calcite-picker-row",
+  styleUrl: "./calcite-picker-row.scss",
+  shadow: true
+})
+export class CalcitePickerRow {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop({ reflect: true }) editing = false;
+
+  @Prop({ reflect: true }) icon: "square" | "circle" | "grip" | null = null;
+
+  @Prop() metadata: object;
+
+  @Prop({ reflect: true, mutable: true }) selected = false;
+
+  @Prop({ reflect: true }) textHeading: string;
+
+  @Prop({ reflect: true }) textDescription: string;
+
+  @Prop({ reflect: true }) value: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerRowToggled: EventEmitter;
+  @Event() calcitePickerRowDeleted: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggle(shiftPressed: boolean) {
+    this.selected = !this.selected;
+    this.calcitePickerRowToggled.emit({
```

Comment:
Would it be better to `@Watch selected`? because if a user sets the selected Property without the toggle method, the event wont be fired

Reply:
Is that a legit use case? If a user is setting selected, then they already have a list of items. Would they want the event to fire in that scenario - giving them the updated internal list if they have a master list they are using to set the state?

Reply:
It seems there is a disconnect between `toggle` and `selected`. They don't do the same thing but they are kind of acting together. If we want to watch the `selected` attribute and make that the source of truth then the event should probably be named `calcitePickerSelectedChange` or something similar. 

It seems like the `toggle` method was introduced just to handle which key was pressed but I think that logic belongs in the Picker widget itself and not on the `row/item`

Reply:
@asangma @kerryrobinson can you guys help define the requirements? Do our dev users need the ability to toggle the selected attribute, or will a public method suffice? There's currently a technical challenge with watching the property and being able to support the shift-click functionality. We'll discuss internally if there's a way to get **click**, **shift+click** as well as **property change** behavior all working together, but if there's not a simple solution, are we ok to proceed with having **click**, **shift+click** and **public method** as ways of interacting with the picker rows?

Reply:
Found a solution. Watching the selected prop has been added.